src/main/java/eu/ai4work/sws/service/RuleEngineService.java

```diff
@@ -31,29 +31,28 @@ public class RuleEngineService {
      * @return SlidingDecision containing the result and the explanation of the sliding decision.
      */
     public SlidingDecision applySlidingDecisionRules(Map<String, Object> slidingDecisionInputParameters) {
-        verifySlidingDecisionInputParameters(fuzzyInferenceSystem, slidingDecisionInputParameters);
+        verifySlidingDecisionInputParameters(slidingDecisionInputParameters);
 
-        setInputParametersToFuzzyInferenceSystem(fuzzyInferenceSystem, slidingDecisionInputParameters);
+        setInputParametersToFuzzyInferenceSystem(slidingDecisionInputParameters);
 
         fuzzyInferenceSystem.evaluate();
 
         String resultAsLinguisticTerm = mapFuzzyInferenceResultToLinguisticTerm(fuzzyInferenceSystem.getVariable(SUGGESTED_WORK_SHARING_APPROACH));
 
-        SlidingDecisionExplanation decisionExplanation = createSlidingDecisionExplanation(fuzzyInferenceSystem);
+        SlidingDecisionExplanation decisionExplanation = createSlidingDecisionExplanation();
```

Comment:
As the parameter is missing now, the code may be easier to read if you rename the method to indicate where the decision comes from, e.g. as follows:
```suggestion
        SlidingDecisionExplanation decisionExplanation = readSlidingDecisionExplanationFromFuzzyInferenceSystem();
```